### PR TITLE
[code42-636] Code42 Integration

### PR DESCRIPTION
**Pull request by nathanhunstad**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
This is the Demisto Code42 integration, including documentation, integration code, unit tests, incident fields, incident type, custom layout, playbooks, and integration test playbook.

## Minimum version of Demisto
- [ ] 4.5.0
- [X] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation 

## Demisto Partner?
- [X] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/Code42/Playbooks/playbook-Code42_Exfiltration_Playbook.yml [hunk 1]
```diff
@@ -0,0 +1,1157 @@
...
+  description: Jira Project for created incident ticket
+- key: JiraType
+  value: {}
+  required: false
+  description: Type of Jira ticket to create
+- key: JiraSummary
+  value: {}
+  required: false
+  description: Summary to use with Jira ticket creation
+outputs: []
```
Comment (yaakovi) [inline Packs/Code42/Playbooks/playbook-Code42_Exfiltration_Playbook.yml]:
```suggestion
outputs: []
tests:
- No Test
```

File: Packs/Code42/IncidentFields/incidentfield-Exfiltrated_Files.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+	"associatedToAll": true,
```
Comment (yaakovi) [inline Packs/Code42/IncidentFields/incidentfield-Exfiltrated_Files.json]: field should be associated to the new incident type and not to all types.
also field should contain `"fromVersion": "5.0.0"`

same for all fields

**Comment by idovandijk** [comment]
Please let me know when you finish with the changes currently requested so I can start my review

File: Packs/Code42/Integrations/Code42/Code42.yml [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      description: Hostname where file event was captured
+      type: string
+    description: Search for a file in Security Data by JSON query, hash, username,
+      device hostname, exfiltration type, or a combination of parameters. At least
+      one parameter must be passed to the command. If a JSON parameter is passed,
+      it will be used to the exclusion of other parameters, otherwise parameters will
+      be combined with an AND clause.
+  - name: code42-alert-get
+    arguments:
+    - name: id
```
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.yml]: This is the alert ID, it is retrieved when fetching alerts via fetch-incidents.

File: Packs/Code42/Integrations/Code42/Code42.yml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      description: Username to remove from the Departing Employee Lens
+    outputs:
+    - contextPath: Code42.DepartingEmployee.CaseID
+      description: Internal Code42 Case ID for Departing Employee
+    - contextPath: Code42.DepartingEmployee.Username
+      description: Username for Departing Employee
+    description: Remove a user from the Departing Employee Lens
+  - name: code42-alert-resolve
+    arguments:
+    - name: id
```
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.yml]: This is the alert ID, it is retrieved when fetching alerts via fetch-incidents.

File: Packs/Code42/Integrations/Code42/Code42.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.yml]: I have updated the description to make it clear that this uses an ID from fetch-incidents.

File: Packs/Code42/Integrations/Code42/Code42.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.yml]: I have updated the description to make it clear that this uses an ID from fetch-incidents.

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+            demisto.incidents(incidents)
+            # Store remaining incidents in integration context
+            integration_context['remaining_incidents'] = remaining_incidents
+            demisto.setIntegrationContext(integration_context)
+        elif command in commands:
+            return_outputs(*commands[command](client, demisto.args()))
+    # Log exceptions
+    except Exception as e:
+        LOG(str(e))
+        LOG.print_log()
```
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.py]: Removed from code.

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+    else:
+        return 'No results found', {}, {}
+
+
+def test_module(client):
+    user = client.get_current_user()
+    if user:
+        return 'ok'
+    else:
+        return 'Error in get_current_user'
```
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.py]: Currently Py42 does not differentiate between different failure modes. I have requested that this be added to Py42. In the meantime, I have updated the error message to be more explicit and match the Py42 error message.

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+    alert = client.resolve_alert(args['id'])
+    if alert:
+        alert_context = {
+            'ID': args['id']
+        }
+        readable_outputs = tableToMarkdown(
+            f'Code42 Security Alert Resolved',
+            alert_context
+        )
+        return readable_outputs, {'Code42.SecurityAlert': alert_context}, alert
```
[the remark this answers is not in the document]
Comment (nathanhunstad) [reply Packs/Code42/Integrations/Code42/Code42.py]: I've updated the call to retrieve the new alert details once it has been resolved.

**Comment by kirbles19** [comment]
@yaakovi Done

**Comment by idovandijk** [comment]
@yaron-libman Finished my review. You can proceed to docs.

**Pull request merged**